### ISSUE TITLE
API: List Pods for a given Deployment

### ISSUE THREAD

**Issue by SanjayVas** [kubernetes/kubernetes#113017]
### What would you like to be added?

Add or update an API method to be able to list Pods for a given deployment.

This is related to #72794, but that focuses on kubectl rather than the underlying API.

### Why is this needed?

The closest that current methods can get is using a [`labelSelector`](https://kubernetes.io/docs/reference/kubernetes-api/common-parameters/common-parameters/#labelSelector) string, but Deployment objects don't contain a single selector string. Instead, the DeploymentSpec has a `selector` field of type [`LabelSelector`](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/label-selector/#LabelSelector), which has `matchLabels` and `matchExpressions` mappings.

Basically, there is no simple way to determine the parameters for a list pods request from a Deployment.

**Comment by lixiaobing1** [comment]
I agree, it can be added to "kubectl describe deployment xxx -n yyy"

**Comment by Gerardo-Salazar** [comment]
This would be very helpful for scanning a Deployment's Pods for status, fields, etc.
